src/lib/components/molecules/slope-chart/index.jsx

```diff
@@ -0,0 +1,110 @@
+import React, { useMemo, useRef, useState, useLayoutEffect } from 'react'
+import { positionLabels, scaleLinear } from './slope-chart-util'
+
+export const SlopeChart = ({
+  id,
+  slopeChartItems,
+  padding = { left: 20, right: 24, top: 20, bottom: 16 },
+  domain,
+  prevValKey,
+  nextValKey,
+  abbrKey,
+  nextLabel,
+  prevLabel,
+}) => {
+  const wrapperRef = useRef(null)
+  const [width, setWidth] = useState(0)
+  const contentSize = Math.floor(width - padding.left - padding.right)
+  const height = contentSize + padding.top + padding.bottom
+  const yScale = scaleLinear(domain, [contentSize, 0])
+  const show = width > 0
+
+  useLayoutEffect(() => {
+    if (width === 0) {
+      setWidth(wrapperRef.current.getBoundingClientRect().width)
+    }
+
+    const handleResize = () => {
+      const newWidth = wrapperRef.current.getBoundingClientRect().width
+      if (newWidth !== width) {
+        setWidth(newWidth)
+      }
+    }
+
+    window.addEventListener('resize', handleResize)
+    return () => window.removeEventListener('resize', handleResize)
+  }, [width])
+
+  const labeLeftPositions = useMemo(
+    () => positionLabels(slopeChartItems, prevValKey, yScale, abbrKey, padding.left - 8, padding.top),
+    [width, slopeChartItems],
+  )
+  const labelRightPositions = useMemo(
+    () => positionLabels(slopeChartItems, nextValKey, yScale, abbrKey, padding.left + contentSize + 8, padding.top),
+    [width, slopeChartItems],
+  )
+
+  return (
+    <div class="gv-slope-chart" ref={wrapperRef}>
```

Comment:
Let's avoid adding class names for now, while we are still figuring out the best styling solution.